packages/orchestrator/internal/template/build/phases/base/hash.go

```diff
@@ -23,9 +25,19 @@ func (bb *BaseBuilder) Hash(_ phases.LayerResult) (string, error) {
 		baseSource = bb.Config.FromImage
 	}
 
+	provisionVersion := provisionScriptFile
+	if val, err := bb.featureFlags.IntFlag(
+		ctx,
+		featureflags.BuildProvisionVersion,
+		featureflags.TemplateContext(bb.Config.TemplateID),
+		featureflags.TeamContext(bb.Config.TeamID, bb.Config.TeamID),
```

Comment:
Might want to make `TeamID` optional here, pass `""` or make a `TeamIDContext` that just takes a TeamID without the name, something like that.